tests/posix/common/src/pthread.c

```diff
@@ -214,9 +216,10 @@ void *thread_top_term(void *p1)
 		}
 	}
 
-	printk("Cancelling thread %d\n", (s32_t) p1);
+	printk("Cancelling thread %d\n", (s32_t) POINTER_TO_INT(p1));
```

Comment:
could just do %p and drop all the casts, here and elsewhere in this test

Reply:
I didn't quite take this approach, but I've cleaned up the tests to reduce the number of casts.